rasa/core/nlg/interpolator.py

```diff
@@ -15,6 +15,15 @@ def interpolate_text(template: Text, values: Dict[Text, Text]) -> Text:
     # (b) { or } in slot name
     try:
         text = re.sub(r"{([^\n{}]+?)}", r"{0[\1]}", template)
+        # for button template, we need to handle escapable curly brace, e.g., button template has
```

Comment:
@8key Tough problem 😬 Actually you can workaround this by double escaping the entity.

See this example
```
def test_interpolate_text():
    text = '/greet{{"entity": "{test}"}}'
    values = {"test": "hello"}
    result = interpolate_text(text, values)

    assert result == '/greet{"entity": "hello"}'
```

Reply:
@wochinge That's why i had added line 26 to handle this. I trust it's good to merge

Reply:
Sorry, I think my message wasn't particularly clear:

You don't need the change in line 26 as you can double escape things which shouldn't be interpolated in your `domain.yml` by double escaping them, e.g.

```
responses:
  utter_greet:
  - buttons:
    - payload: '/greet{{"some_entity": "{test}"}}'
      title: Hi {test}
```